src/reqs.c

```diff
@@ -369,12 +369,15 @@ static struct request_s *process_request (struct conn_s *connptr,
 
                 reverse_url = reverse_rewrite_url (connptr, hashofheaders, url);
 
-                if (!reverse_url) {
+                if (reverse_url != NULL) {
+                        safefree (url);
+                        url = reverse_url;
+		} else if (config.reverseonly) {
+                        indicate_http_error (connptr, 400, "Bad Request",
+                                     "detail", "No mapping found for requested url",
```

Comment:
Formatting: please indent the arguments to match the opening bracket, and make sure the line length does not exceed 80 characters, if possible. i.e.

```
    func (connptr, ...
          "detail", "No ...."
          ...);
```